### PR TITLE
feat(bundler): support custom sign command on Windows

### PR DESCRIPTION
**Pull request by muwoo**
feat: Migrate v2 functionality to v1.x: support custom sign command on Windows (Discussed in https://github.com/tauri-apps/tauri/discussions/9900)

- [x] cargo test passes
- [x] cargo clippy passes

### CODE REVIEW

**Review by amrbashir — APPROVED** [review]

**Review by amrbashir — CHANGES_REQUESTED** [review]
Looks like you need to run `cargo fmt` and `cargo clippy -- -Dwarnings`

**Comment by muwoo** [comment]
> Looks like you need to run `cargo fmt` and `cargo clippy -- -Dwarnings`

done

**Review by amrbashir — APPROVED** [review]

File: tooling/bundler/src/bundle/windows/nsis.rs [hunk 1]
```diff
@@ -68,6 +67,7 @@ pub fn bundle_project(settings: &Settings, updater: bool) -> crate::Result<Vec<P
   let nsis_toolset_path = tauri_tools_path.join("NSIS");
 
   if !nsis_toolset_path.exists() {
+    create_dir_all(&nsis_toolset_path)?;
```
Comment (MystiPanda) [inline tooling/bundler/src/bundle/windows/nsis.rs:70]: Creating an NSIS folder in advance will cause subsequent renaming failures after extract

```rust
rename(_tauri_tools_path.join("nsis-3.08"), nsis_toolset_path)?;
```

**Review by MystiPanda — COMMENTED** [review]
Maybe it needs to be modified here.

@amrbashir Could you please take a look at it?

**Comment by tr3ysmith** [comment]
@amrbashir any thoughts on when v1.7 would be released that contains this fix?

**Comment by amrbashir** [comment]
I don't want to make promises but hopefully this week

**Comment by tr3ysmith** [comment]
@amrbashir I see 1.7 came out, but I don't see anything about this feature in the release notes, is it indeed in there?

**Comment by amrbashir** [comment]
@tr3ysmith here it is https://github.com/tauri-apps/tauri/releases/tag/tauri-bundler-v1.6.0

**Comment by ttrushin** [comment]
Maybe a silly question, but I have tauri updated to 1.7.1 (and tauri-build updated to 1.5.3) so everything appears to be updated to latest, but I cannot build with "signCommand". 

```Error `tauri.conf.json` error on `tauri > bundle > windows`: Additional properties are not allowed ('signCommand' was unexpected)```

Anyone know why?

**Comment by amrbashir** [comment]
make sure to also update the CLI

**Pull request merged**